### PR TITLE
Convert EDD plugins from POST to GET

### LINKED ISSUE #27: Failed to execute (9) unzip -qq -o 'web/app/plugins/wp-all-import-pro// ...
Hey all,

Trying to use the all import/export pro plugin part of this.

Seems that I am getting a similar error as the issues referenced below (and commented in issue 18):
https://github.com/junaidbhura/composer-wp-pro-plugins/issues/18
https://github.com/junaidbhura/composer-wp-pro-plugins/issues/14
https://github.com/junaidbhura/composer-wp-pro-plugins/issues/1

Screenshots of my error:
Screen Shot 2021-02-24 at 12 40 46 PM

Text version:
```
[RuntimeException]                                                                                     
  Failed to execute (9) unzip -qq -o 'web/app/plugins/wp-all-import-pro//7d7969993593a13942741e4a52c3e8  
  29' -d '/Users/jesstucker/Code/redacted/site/vendor/composer/b8cc9703'                  
                                                                                                         
  [web/app/plugins/wp-all-import-pro//7d7969993593a13942741e4a52c3e829]                                  
    End-of-central-directory signature not found.  Either this file is not                               
    a zipfile, or it constitutes one disk of a multi-part archive.  In the                               
    latter case the central directory and zipfile comment will be found on                               
    the last disk(s) of this archive.                                                                    
  unzip:  cannot find zipfile directory in one of web/app/plugins/wp-all-import-pro//7d7969993593a13942  
  741e4a52c3e829 or                                                                                      
          web/app/plugins/wp-all-import-pro//7d7969993593a13942741e4a52c3e829.zip, and cannot find web/  
  app/plugins/wp-all-import-pro//7d7969993593a13942741e4a52c3e829.ZIP, period.   
```

I noticed that some previous fixes had occurred on `Installer.php` but I must be honest in admitting I am terribly new to composer / wp / etc... and not really sure where / what to inspect to catch the error.

@junaidbhura asked me to post my versions in the prior post; here they are again:

`composer.json`:

```
...
{
      "type": "package",
      "package": {
        "name": "junaidbhura/wp-all-import-pro",
        "version": "4.6.5",
        "type": "wordpress-plugin",
        "dist": {
          "type": "zip",
          "url": "https://www.wpallimport.com"
        },
        "require": {
          "junaidbhura/composer-wp-pro-plugins": "*"
        }
      }
    },
    {
      "type": "package",
      "package": {
        "name": "junaidbhura/wp-all-export-pro",
        "version": "1.6.4",
        "type": "wordpress-plugin",
        "dist": {
          "type": "zip",
          "url": "https://www.wpallimport.com"
        },
        "require": {
          "junaidbhura/composer-wp-pro-plugins": "*"
        }
      }
    }
...
 "require": {
    ...
    "junaidbhura/wp-all-import-pro": "*",
    "junaidbhura/wp-all-export-pro": "*"
  },
```

Perhaps I will also add that I am using Bedrock / Roots / Trellis / Sage 10
Also my assumption is that I am using the 1.2.0 version of this plugin (composer-wp-pro-plugins)

Love what you're doing, managing these paid plugins is a headache!

Sincerely,
Jess

### PR DESCRIPTION
**Pull request by jessedyck**
## Checklist:
- [x] I've read the [Contributing page](https://github.com/junaidbhura/composer-wp-pro-plugins/blob/master/CONTRIBUTING.md).
- [x] I've created an issue and referenced it here.
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [ ] My code has proper inline documentation.

## Description
Updated EDD plugins, Polylang and WPAll Import, to use GET requests when calling the software licensing API to get the URL of the download. Fixes https://github.com/junaidbhura/composer-wp-pro-plugins/issues/27

## How has this been tested?
✅ Tested installing WP All Import with valid key
🚫 Polylang was not tested - I don't have a license.  Would be great to get that tested prior to merging. 

## Types of changes
* Bug fix

### CODE REVIEW

File: plugins/PolylangPro.php [hunk 1]
```diff
@@ -37,17 +37,20 @@ public function __construct( $version = '' ) {
 	 */
 	public function getDownloadUrl() {
 		$http     = new Http();
-		$response = json_decode( $http->post( 'https://polylang.pro', array(
+		$name = 'Polylang Pro';
```
Comment (junaidbhura) [inline plugins/PolylangPro.php:40]: @jessedyck Can you undo the changes in this file? Nobody has faced any issues with Polylang, and I don't want to inadvertently cause any issues. WP All Import / Export seem to have a weird EDD implementation.

File: plugins/WpAiPro.php [hunk 2]
```diff
@@ -100,8 +101,9 @@ public function getDownloadUrl() {
 		) ), true );
 		if ( ! empty( $response['download_link'] ) ) {
 			return $response['download_link'];
+		} else {
```
Comment (junaidbhura) [inline plugins/WpAiPro.php:104]: @jessedyck Can we undo this change?

File: plugins/WpAiPro.php [hunk 3]
```diff
@@ -7,6 +7,7 @@
 
 namespace Junaidbhura\Composer\WPProPlugins\Plugins;
 
+use Exception;
```
Comment (junaidbhura) [inline plugins/WpAiPro.php:10]: @jessedyck Can we undo this change?

File: Http.php [hunk 4]
```diff
@@ -36,4 +36,27 @@ public function post( $url = '', $args = array() ) {
...
+	 * GET request.
+	 *
+	 * @param string $url Base URL for requeset (without params)
+	 * @param array  $args Arguments to add to request
+	 * @return mixed
+	 */
+	public function get( $url = '', $args = array() ) {
+		$query_string = '';
+
+		$curl_handle = curl_init();
```
Comment (junaidbhura) [inline Http.php:49]: @jessedyck Have you tested this on Windows? I remember facing an issue with the POST request.

**Review by junaidbhura — CHANGES_REQUESTED** [review]

**Comment by junaidbhura** [comment]
Thanks a bunch for your PR, @jessedyck !

Could you please elaborate in what environments / Operating Systems you've tested your changes?

**Comment by junaidbhura** [comment]
@jessedyck I've merged this PR and made the updates. However, it looks like WP All Import / Export have recently started using Cloudflare, so we're unable to download from the command line at the moment, so I've not created a release just yet!

**Comment by jessedyck** [comment]
That may explain a change in behaviour I noticed recently; any request would return a download link, regardless of license key. That doesn’t match the expected behaviour from the EDD docs though. I haven’t had a chance to revisit, but perhaps it was cloudflare caching the response then. 

I’ll try to take a peak soon!

**Comment by junaidbhura** [comment]
That's odd @jessedyck I went ahead and did a release: https://github.com/junaidbhura/composer-wp-pro-plugins/releases/tag/1.5.0 hoping to see if I'd have luck in CI environments, but no dice!

I hope they fix the Cloudflare thing soon.

**Pull request merged**